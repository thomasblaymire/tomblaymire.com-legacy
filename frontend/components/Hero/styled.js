import styled from 'styled-components';
import heroImage from '../../static/me.jpg';
import stock from '../../static/stock.jpg';

export const StyledHero = styled.div`
  background-image: url(${stock});
  background-size: cover;
  min-height: 30vh;
  height: auto;
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: fixed;
  position: relative;
  background-position-x: -500px;

  @media (min-width: 480px) {
    min-height: 80vh;
    background-position-x: center;
  }
`;

export const StyledHeroWrapper = styled.div`
  max-width: 1170px;
  margin-right: auto;
  margin-left: auto;
  padding: 0px 15px;
`;

export const StyledHeroContent = styled.div`
  color: ${(props) => props.theme.white};
  width: 100%;
  padding: 22rem 0;
  line-height: 20px;
  @media (min-width: 480px) {
    width: 60%;
  }
`;

export const StyledHeroTitle = styled.h2`
  font-size: 6rem;
`;

export const StyledHeroText = styled.p`
  font-size: 2.2rem;
  line-height: 3.5em;
  font-weight: 300;
  padding-bottom: 5rem;
  @media (min-width: 480px) {
    line-height: 1.5em;
    font-weight: 300;
    min-width: 80%;
  }
`;

export const StyledButtonBlock = styled.div`
  display: flex;

  & > button {
    margin-right: 1.8rem;
  }

  a {
    color: ${(props) => props.theme.white};
    text-decoration: none;
    font-weight: normal;
    &:hover {
      color: red;
    }
  }
`;
