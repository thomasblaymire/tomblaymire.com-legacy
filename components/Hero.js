import styled from 'styled-components';
import Link from 'next/link';
import hero from '../static/me.jpg';
import Button from './Button';

const HeroImage = styled.div`
	background-image: url(${hero});
	background-size: cover;
	min-height: 100vh;
`;

const HeroContainer = styled.div`
	width: 1170px;
	margin-right: auto;
	margin-left: auto;
	padding-left: 15px;
	padding-right: 15px;
`;

const HeroBlock = styled.div`
	color: #fff;
	margin-top: 11rem;
	width: 60%;

	.intro {
		font-size: 5.5rem;
		margin: 0;
	}

	.description {
		font-size: 2.5rem;
		line-height: 38px;
		font-weight: 300;
	}
`;

const HeroButtons = styled.div`
	margin-top: 8rem;
	display: flex;

	& > * {
		padding-right: 2rem;
	}

	a {
		color: #ffffff;
		text-decoration: none;

		font-weight: normal;
	}
`;

const Hero = props => {
	return (
		<HeroImage>
			<HeroContainer>
				{props.children}
				<HeroBlock>
					<h2 className="intro">Hello, I'm Tom</h2>
					<div className="description">
						Iâ€™m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end
						products that meet both user and business needs.
					</div>
					<HeroButtons>
						<Button type="primary">
							<Link href="/portfolio">
								<a>My Work</a>
							</Link>
						</Button>
						<Button type="secondary">
							<Link href="/contact">Let's Talk</Link>
						</Button>
					</HeroButtons>
				</HeroBlock>
			</HeroContainer>
		</HeroImage>
	);
};

export default Hero;
