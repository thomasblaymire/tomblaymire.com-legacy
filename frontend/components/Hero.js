import styled from 'styled-components';
import Link from 'next/link';
import hero from '../static/me.jpg';
import Button from './Button';

const HeroImage = styled.div`
	background-image: url(${hero});
	background-size: cover;
	min-height: 70vh;
	background-repeat: no-repeat;
	background-position: center center;
	background-attachment: fixed;
	position: relative;

	@media (min-width: 480px) {
		min-height: 100vh;
	}
`;

const HeroContainer = styled.div`
	max-width: 1170px;
	margin-right: auto;
	margin-left: auto;
	padding: 0px 15px;
`;

const HeroBlock = styled.div`
	color: ${props => props.theme.white};
	margin-top: 7rem;
	width: 100%;
	position: absolute;
	bottom: 25rem;

	@media (min-width: 480px) {
		width: 60%;
	}

	.intro {
		font-size: 7vmin;
		margin: 0;
	}

	.description {
		font-size: 1.6rem;
		line-height: 2.6rem;
		font-weight: 300;
		@media (min-width: 480px) {
			line-height: 4.5rem;
			font-size: 3.5vmin;
			font-weight: 300;
			min-width: 80%;
		}
	}
`;

const HeroButtons = styled.div`
	margin-top: 8rem;
	display: flex;

	& > button {
		margin-right: 1.8rem;
	}

	a {
		color: ${props => props.theme.white};
		text-decoration: none;
		font-weight: normal;
		&:hover {
			color: red;
		}
	}
`;

const Hero = props => {
	return (
		<HeroImage>
			<HeroContainer>
				{props.children}
				<HeroBlock>
					<h2 className="intro">Hello, I'm Tom</h2>
					<div className="description">
						Iâ€™m a Freelance (UI/UX) Software Engineer based in Sydney. I help companies deliver innovative, end-to-end
						products that meet both user and business needs.
					</div>
					<HeroButtons>
						<Link href="/portfolio">
							<Button primaryColor="#fff" borderColor={'#05af43'} bgColor="#05af43" hoverBg={'#05da53'}>
								My Work
							</Button>
						</Link>
						<Link href="/contact">
							<Button
								primaryColor="#fff"
								bgColor="transparent"
								borderColor={'#fff'}
								hoverBg={'#fff'}
								hoverColor={'#000'}
							>
								Let's Talk
							</Button>
						</Link>
					</HeroButtons>
				</HeroBlock>
			</HeroContainer>
		</HeroImage>
	);
};

export default Hero;
