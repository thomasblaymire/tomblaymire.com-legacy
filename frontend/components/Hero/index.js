import Link from 'next/link';
import Button from '../Button';
import { StyledHero, StyledHeroWrapper, StyledHeroContent, StyledButtonBlock } from './styled';

const Hero = ({ title, description }) => (
  <StyledHero>
    <StyledHeroWrapper>
      {props.children}
      <StyledHeroContent>
        <h2 className="intro">{title}</h2>
        <div className="description">{description}</div>
        <StyledButtonBlock>
          <Link href="/portfolio">
            <Button primaryColor="#fff" borderColor={'#05af43'} bgColor="#05af43" hoverBg={'#05da53'}>
              My Work
            </Button>
          </Link>
          <Link href="/contact">
            <Button
              primaryColor="#fff"
              bgColor="transparent"
              borderColor={'#fff'}
              hoverBg={'#fff'}
              hoverColor={'#000'}
            >
              Let's Talk
            </Button>
          </Link>
        </StyledButtonBlock>
      </StyledHeroContent>
    </StyledHeroWrapper>
  </StyledHero>
);

export default Hero;
