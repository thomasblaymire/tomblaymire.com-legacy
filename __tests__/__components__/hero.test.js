import { shallow } from 'enzyme';
import { Jest } from 'jest';
import * as React from 'react';
import toJson from 'enzyme-to-json';
import Hero from '../../components/Hero';

const renderComponent = (props) => shallow(<Hero {...props} />);

describe('The Hero Component', () => {

    it('Renders the Hero component', () => {
        const renderedComponent = renderComponent({});
        expect(renderedComponent).toMatchSnapshot();
    });

    expect(toJson(renderedComponent)).toMatchSnapshot();

});