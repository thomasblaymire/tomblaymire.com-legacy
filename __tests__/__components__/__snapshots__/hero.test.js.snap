// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Hero Component Renders the Hero component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Hero />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Hero__HeroContainer>
        <Hero__HeroBlock>
          <h2
            className="intro"
          >
            Hello, I'm Tom
          </h2>
          <div
            className="description"
          >
            I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
          </div>
          <Hero__HeroButtons>
            <Link
              href="/portfolio"
            >
              <Button
                bgColor="#05af43"
                borderColor="#05af43"
                hoverBg="#05da53"
                primaryColor="#fff"
              >
                My Work
              </Button>
            </Link>
            <Link
              href="/contact"
            >
              <Button
                bgColor="transparent"
                borderColor="#fff"
                hoverBg="#fff"
                hoverColor="#000"
                primaryColor="#fff"
              >
                Let's Talk
              </Button>
            </Link>
          </Hero__HeroButtons>
        </Hero__HeroBlock>
      </Hero__HeroContainer>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          undefined,
          <Hero__HeroBlock>
            <h2
              className="intro"
            >
              Hello, I'm Tom
            </h2>
            <div
              className="description"
            >
              I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
            </div>
            <Hero__HeroButtons>
              <Link
                href="/portfolio"
              >
                <Button
                  bgColor="#05af43"
                  borderColor="#05af43"
                  hoverBg="#05da53"
                  primaryColor="#fff"
                >
                  My Work
                </Button>
              </Link>
              <Link
                href="/contact"
              >
                <Button
                  bgColor="transparent"
                  borderColor="#fff"
                  hoverBg="#fff"
                  hoverColor="#000"
                  primaryColor="#fff"
                >
                  Let's Talk
                </Button>
              </Link>
            </Hero__HeroButtons>
          </Hero__HeroBlock>,
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <h2
                className="intro"
              >
                Hello, I'm Tom
              </h2>,
              <div
                className="description"
              >
                I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
              </div>,
              <Hero__HeroButtons>
                <Link
                  href="/portfolio"
                >
                  <Button
                    bgColor="#05af43"
                    borderColor="#05af43"
                    hoverBg="#05da53"
                    primaryColor="#fff"
                  >
                    My Work
                  </Button>
                </Link>
                <Link
                  href="/contact"
                >
                  <Button
                    bgColor="transparent"
                    borderColor="#fff"
                    hoverBg="#fff"
                    hoverColor="#000"
                    primaryColor="#fff"
                  >
                    Let's Talk
                  </Button>
                </Link>
              </Hero__HeroButtons>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hello, I'm Tom",
                "className": "intro",
              },
              "ref": null,
              "rendered": "Hello, I'm Tom",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.",
                "className": "description",
              },
              "ref": null,
              "rendered": "I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Link
                    href="/portfolio"
                  >
                    <Button
                      bgColor="#05af43"
                      borderColor="#05af43"
                      hoverBg="#05da53"
                      primaryColor="#fff"
                    >
                      My Work
                    </Button>
                  </Link>,
                  <Link
                    href="/contact"
                  >
                    <Button
                      bgColor="transparent"
                      borderColor="#fff"
                      hoverBg="#fff"
                      hoverColor="#000"
                      primaryColor="#fff"
                    >
                      Let's Talk
                    </Button>
                  </Link>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      bgColor="#05af43"
                      borderColor="#05af43"
                      hoverBg="#05da53"
                      primaryColor="#fff"
                    >
                      My Work
                    </Button>,
                    "href": "/portfolio",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bgColor": "#05af43",
                      "borderColor": "#05af43",
                      "children": "My Work",
                      "hoverBg": "#05da53",
                      "primaryColor": "#fff",
                    },
                    "ref": null,
                    "rendered": "My Work",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <Button
                      bgColor="transparent"
                      borderColor="#fff"
                      hoverBg="#fff"
                      hoverColor="#000"
                      primaryColor="#fff"
                    >
                      Let's Talk
                    </Button>,
                    "href": "/contact",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "bgColor": "transparent",
                      "borderColor": "#fff",
                      "children": "Let's Talk",
                      "hoverBg": "#fff",
                      "hoverColor": "#000",
                      "primaryColor": "#fff",
                    },
                    "ref": null,
                    "rendered": "Let's Talk",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Hero__HeroContainer>
          <Hero__HeroBlock>
            <h2
              className="intro"
            >
              Hello, I'm Tom
            </h2>
            <div
              className="description"
            >
              I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
            </div>
            <Hero__HeroButtons>
              <Link
                href="/portfolio"
              >
                <Button
                  bgColor="#05af43"
                  borderColor="#05af43"
                  hoverBg="#05da53"
                  primaryColor="#fff"
                >
                  My Work
                </Button>
              </Link>
              <Link
                href="/contact"
              >
                <Button
                  bgColor="transparent"
                  borderColor="#fff"
                  hoverBg="#fff"
                  hoverColor="#000"
                  primaryColor="#fff"
                >
                  Let's Talk
                </Button>
              </Link>
            </Hero__HeroButtons>
          </Hero__HeroBlock>
        </Hero__HeroContainer>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            undefined,
            <Hero__HeroBlock>
              <h2
                className="intro"
              >
                Hello, I'm Tom
              </h2>
              <div
                className="description"
              >
                I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
              </div>
              <Hero__HeroButtons>
                <Link
                  href="/portfolio"
                >
                  <Button
                    bgColor="#05af43"
                    borderColor="#05af43"
                    hoverBg="#05da53"
                    primaryColor="#fff"
                  >
                    My Work
                  </Button>
                </Link>
                <Link
                  href="/contact"
                >
                  <Button
                    bgColor="transparent"
                    borderColor="#fff"
                    hoverBg="#fff"
                    hoverColor="#000"
                    primaryColor="#fff"
                  >
                    Let's Talk
                  </Button>
                </Link>
              </Hero__HeroButtons>
            </Hero__HeroBlock>,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <h2
                  className="intro"
                >
                  Hello, I'm Tom
                </h2>,
                <div
                  className="description"
                >
                  I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.
                </div>,
                <Hero__HeroButtons>
                  <Link
                    href="/portfolio"
                  >
                    <Button
                      bgColor="#05af43"
                      borderColor="#05af43"
                      hoverBg="#05da53"
                      primaryColor="#fff"
                    >
                      My Work
                    </Button>
                  </Link>
                  <Link
                    href="/contact"
                  >
                    <Button
                      bgColor="transparent"
                      borderColor="#fff"
                      hoverBg="#fff"
                      hoverColor="#000"
                      primaryColor="#fff"
                    >
                      Let's Talk
                    </Button>
                  </Link>
                </Hero__HeroButtons>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hello, I'm Tom",
                  "className": "intro",
                },
                "ref": null,
                "rendered": "Hello, I'm Tom",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.",
                  "className": "description",
                },
                "ref": null,
                "rendered": "I’m a Freelance (UI/UX) Software Engineer based in Leeds. I help companies deliver innovative, end-to-end products that meet both user and business needs.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Link
                      href="/portfolio"
                    >
                      <Button
                        bgColor="#05af43"
                        borderColor="#05af43"
                        hoverBg="#05da53"
                        primaryColor="#fff"
                      >
                        My Work
                      </Button>
                    </Link>,
                    <Link
                      href="/contact"
                    >
                      <Button
                        bgColor="transparent"
                        borderColor="#fff"
                        hoverBg="#fff"
                        hoverColor="#000"
                        primaryColor="#fff"
                      >
                        Let's Talk
                      </Button>
                    </Link>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Button
                        bgColor="#05af43"
                        borderColor="#05af43"
                        hoverBg="#05da53"
                        primaryColor="#fff"
                      >
                        My Work
                      </Button>,
                      "href": "/portfolio",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "bgColor": "#05af43",
                        "borderColor": "#05af43",
                        "children": "My Work",
                        "hoverBg": "#05da53",
                        "primaryColor": "#fff",
                      },
                      "ref": null,
                      "rendered": "My Work",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": <Button
                        bgColor="transparent"
                        borderColor="#fff"
                        hoverBg="#fff"
                        hoverColor="#000"
                        primaryColor="#fff"
                      >
                        Let's Talk
                      </Button>,
                      "href": "/contact",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "bgColor": "transparent",
                        "borderColor": "#fff",
                        "children": "Let's Talk",
                        "hoverBg": "#fff",
                        "hoverColor": "#000",
                        "primaryColor": "#fff",
                      },
                      "ref": null,
                      "rendered": "Let's Talk",
                      "type": [Function],
                    },
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
